pkgs/applications/editors/vscode-oss/default.nix

```diff
@@ -0,0 +1,214 @@
+{ stdenv, fetchFromGitHub, python, zip, makeWrapper, fetchurl, pkgconfig, jq
+, makeDesktopItem, runtimeShell, writeScript
+, nodejs, yarn, libsecret, xorg, ripgrep, electron
+, productOverride ? {}
+}:
+
+let
+  shortName = productOverride.nameShort or "Code - OSS";
+  longName = productOverride.nameLong or "Code - OSS";
+  executableName = productOverride.applicationName or "code-oss";
+
+  vscodePlatforms = {
+    i686-linux = "linux-ia32";
+    x86_64-linux = "linux-x64";
+    armv7l-linux = "linux-armv7l";
+    aarch64-linux = "linux-arm64";
+    x86_64-darwin = "darwin-x64";
+  };
+
+  system = stdenv.hostPlatform.system;
+
+  platform = vscodePlatforms.${system} or (throw "Unsupported platform: ${system}");
+
+in stdenv.mkDerivation rec {
+  pname = "vscode";
+  version = "1.44.2";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "vscode";
+    rev = version;
+    sha256 = "z0XdyD3lpcL90uGFlHeWHEJnnXmYPfpeaU0e2xjJnGU=";
+  };
+
+  yarnCache = stdenv.mkDerivation {
+    name = "${pname}-${version}-yarn-cache";
+    inherit src;
+    phases = ["unpackPhase" "buildPhase"];
+    nativeBuildInputs = [ yarn ];
+    buildPhase = ''
+      export HOME=$PWD
+
+      yarn config set yarn-offline-mirror $out
+
+      find . -name "yarn.lock" \
+        -execdir yarn install --frozen-lockfile --ignore-scripts --no-progress --non-interactive \;
+    '';
+    outputHashMode = "recursive";
+    outputHashAlgo = "sha256";
+    outputHash = "hX15YUCgG5RAAjit/UnBVvzm1TCfHMPe0WqJYkV1DDo=";
+  };
+
+  productOverrideJSON = builtins.toFile "product-override.json" (builtins.toJSON productOverride);
+
+  nativeBuildInputs = [ nodejs yarn python pkgconfig zip makeWrapper jq ];
+  buildInputs = [ libsecret xorg.libX11 xorg.libxkbfile ];
+
+  BUILD_SOURCEVERSION = version;
+
+  desktopItem = makeDesktopItem {
+    name = executableName;
+    desktopName = longName;
+    comment = "Code Editing. Redefined.";
+    genericName = "Text Editor";
+    exec = executableName;
+    icon = "code";
+    startupNotify = "true";
+    categories = "Utility;TextEditor;Development;IDE;";
+    mimeType = "text/plain;inode/directory;";
+    extraEntries = ''
+      StartupWMClass=${shortName}
+      Actions=new-empty-window;
+      Keywords=vscode;
+
+      [Desktop Action new-empty-window]
+      Name=New Empty Window
+      Exec=${executableName} --new-window %F
+      Icon=code
+    '';
+  };
+
+  urlHandlerDesktopItem = makeDesktopItem {
+    name = executableName + "-url-handler";
+    desktopName = longName + " - URL Handler";
+    comment = "Code Editing. Redefined.";
+    genericName = "Text Editor";
+    exec = executableName + " --open-url %U";
+    icon = "code";
+    startupNotify = "true";
+    categories = "Utility;TextEditor;Development;IDE;";
+    mimeType = "x-scheme-handler/vscode;";
+    extraEntries = ''
+      NoDisplay=true
+      Keywords=vscode;
+    '';
+  };
+
+  electronWrapper = writeScript "electron" ''
```

Comment:
Shouldn't this be just `makeWrapper` call in the `postFixup`?

Similar to https://github.com/NixOS/nixpkgs/blob/5f814bf9cc466be415d784a9cbf0fbae65428c24/pkgs/applications/networking/instant-messengers/jitsi-meet-electron/default.nix#L44-L48

Reply:
I need additional `if` condition, and this cannot be done by `makeWrapper` as far as I know.

Reply:
There is a `--run` flag but it would get quite mess here given the complexity of the script.

Reply:
Ok, let's keep it with custom script.